tst/EnergyPlus/unit/IndoorGreen.unit.cc

```diff
@@ -0,0 +1,227 @@
+        "    Fraction,                !- Schedule Type Limits Name",
+        "    Through: 12/31,          !- Field 1",
+        "    For: AllDays,            !- Field 2",
+        "    Until: 24:00,            !- Field 3",
+        "    0.0;                     !- Field 4",
+
+        "  IndoorLivingWall,",
+        "    Space1-1IndoorLivingWall, !-Name",
+        "    SPACE1-1SouthPartition, !-Surface Name",
+        "    AlwaysOn,                   !-Schedule Name",
+        "    Penman-Monteith,            !-ET Calculation Method",
+        "    LED,                        !-Lighting Method",
+        "    AlwaysOff,                  !-LED Intensity Schedule Name",
+        "    ,                           !-Daylighting Control Name",
+        "    ,                           !-LED - Daylight Targeted Lighting Intensity Schedule Name",
+        "    30,                         !- Total Leaf Area {m2}",
+        "    32.5,                       !- LED Nominal Intensity {umol_m2s}",
+        "    640,                        !- LED Nominal Power {W}",
+        "    0.6;                        !- Radiant Fraction of LED Lights",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    Material::GetMaterialData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    HeatBalanceManager::GetConstructData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    HeatBalanceManager::GetZoneData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    state->dataSurfaceGeometry->CosZoneRelNorth.allocate(1);
+    state->dataSurfaceGeometry->SinZoneRelNorth.allocate(1);
+    state->dataSurfaceGeometry->CosZoneRelNorth(1) = std::cos(-state->dataHeatBal->Zone(1).RelNorth * Constant::DegToRadians);
+    state->dataSurfaceGeometry->SinZoneRelNorth(1) = std::sin(-state->dataHeatBal->Zone(1).RelNorth * Constant::DegToRadians);
+    state->dataSurfaceGeometry->CosBldgRelNorth = 1.0;
+    state->dataSurfaceGeometry->SinBldgRelNorth = 0.0;
+
+    SurfaceGeometry::GetSurfaceData(*state, ErrorsFound);
+    EXPECT_FALSE(ErrorsFound);
+
+    state->dataHVACGlobal->TimeStepSys = 1;
+    state->dataHVACGlobal->TimeStepSysSec = state->dataHVACGlobal->TimeStepSys * Constant::SecInHour;
+
+    state->dataGlobal->NumOfTimeStepInHour = 1;
+    state->dataGlobal->MinutesPerTimeStep = 60 / state->dataGlobal->NumOfTimeStepInHour;
+    state->dataGlobal->TimeStep = 1;
+    state->dataGlobal->HourOfDay = 1;
+    state->dataEnvrn->DayOfWeek = 1;
+    state->dataEnvrn->DayOfYear_Schedule = 1;
+    ScheduleManager::UpdateScheduleValues(*state);
+
+    IndoorGreen::GetIndoorGreenInput(*state, ErrorsFound); // setup indoor living wall data
+    EXPECT_FALSE(ErrorsFound);                             // expect no errors
+    auto &thisindoorgreen = state->dataIndoorGreen->indoorGreens(IndoorGreenNum);
+    EXPECT_EQ(thisindoorgreen.LeafArea, 30.0);
+    EXPECT_EQ(thisindoorgreen.lightingMethod, IndoorGreen::LightingMethod::LED);
```

Comment:
There are some [build errors](https://myoldmopar.github.io/EnergyPlusBuildResults/EnergyPlus-a97bf77897f2e8cd988a88a7573ea040e6664c89-PR10385-x86_64-Linux-Ubuntu-22.04-gcc-11.4-UnitTestsCoverage-Debug.html). For enums use 
`EXPECT_TRUE(compare_enums(thisindoorgreen.lightingMethod, IndoorGreen::LightingMethod::LED));`

Reply:
I'm not sure whether @Myoldmopar would co-sign on this, but my preferred idiom is:

```c++
EXPECT_EQ((int)thisindoorgreen.lightingMethod, (int)IndoorGreen::LightingMethod::LED);
```

I don't know how many universes there are in the multi-verse, but I am pretty sure that in all of them `enum` is implemented as `int`.

Reply:
Thank you so much! I added both enum test functions for the unit test. Works well!

Reply:
I would co-sign on either.  `compare_enums` is what we have in all the code, but it's doing the same thing.  I have a different enum thing to propose/discuss aside from this, so I'll just say this is fine.